### DISCUSSION TITLE
I'm having trouble fitting a subquery pattern into the system

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#3249] [Q&A] [answered]
This sql is coming out of a functioning view collection.  It works, except that the user id is hard coded in.  This is for a common members suggestion when adding other users to a site.  It looks for all users which share a site with a user, and then it returns all users which also share one of those sites with the main user.

The problem is figuring out how to add in parameters to a view collection.  Do I need to go directly to SQL with any ORM layer for something like this?

```sql
SELECT
  (ROW_NUMBER() OVER()) as id,
  su.user as user,
  u.email as email,
  u.name  as name
FROM (
  SELECT site FROM sites_users su
  WHERE su.user = 'fln1dub00601tf1'
) as common

LEFT JOIN sites_users su on su.site = common.site

LEFT JOIN users u on u.id = su.user

WHERE su.user <> 'fln1dub00601tf1'
```

**Comment by dschissler** [comment]
Oof.  I can't figure out how to do this, unless I return a cartesian product in the results so that it can be filtered in the API rules.  I also can't figure out how to do this search in dao or dbx.

**Reply by dschissler** [reply]
@ganigeorgiev Is there any easier way to get a vector out of dbx than this?

```golang
	var res []dbx.NullStringMap
	err = dao.DB().
		Select("user").
		From(models.SitesUsers).
		Where(dbx.HashExp{
			"sites_users.site": siteId,
		}).
		All(&res)

	siteIdArr := []string{}
	for _, item := range res {
		siteIdArr = append(siteIdArr, item["user"].String)
	}
	fmt.Println(siteIdArr)

```

**Reply by ganigeorgiev** [reply, marked as answer]
You can use a slice of struct to scan the result in instead of `[]dbx.NullStringMap` (for example see https://pocketbase.io/docs/go-database/#execute-all).
Or if you just want the user column results you can use [`Column()`](https://pkg.go.dev/github.com/pocketbase/dbx#SelectQuery.Column), something like:

```go
var users []string
err = dao.DB().
    Select("user").
    From(models.SitesUsers).
    Where(dbx.HashExp{
        "sites_users.site": siteId,
    }).
    Column(&users)
```

Your initial query can be also rougly written as (_I haven't tested it_):

```go
type Member struct {
    Id    string
    Email string
    Name  string
}

var members []Member

err := app.Dao().DB().
    NewQuery(`
        SELECT DISTINCT
          su.user as id,
          u.email as email,
          u.name  as name
        FROM (
          SELECT site FROM sites_users su
          WHERE su.user = {:userId}
        ) as common
        LEFT JOIN sites_users su on su.site = common.site
        LEFT JOIN users u on u.id = su.user
        WHERE su.user <> {:userId}
        LIMIT 100
    `).
    Bind(dbx.Params{
        "userId": userId,
    }).
    All(&members)
```

**Reply by dschissler** [reply]
@ganigeorgiev Thanks!

1) Damn, I didn't know that the `.Column` method exists.

2) Damn, I didn't realize that a `.NewQuery` can be made from the Builder.

dbx is pretty nice, but the documentation is bad.  I think that there are only 4 examples for the entire project.  Then I suppose there are tests.

Also, I don't know what is going wrong with the VS Code to Golang package docs, because it never seems to arrive at the correct place on the website.  It seems broken.  In total a bad experience.

So dbx released a version in like 3 years.  Maybe you'll have to take over by having the dominant fork of the project.
